x-pack/plugins/actions/server/builtin_action_types/torq.test.ts

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Services } from '../types';
+import { validateConfig, validateSecrets, validateParams } from '../lib';
+import { actionsConfigMock } from '../actions_config.mock';
+import { createActionTypeRegistry } from './index.test';
+import { Logger } from '@kbn/core/server';
+import { actionsMock } from '../mocks';
+import axios from 'axios';
+import {
+  ActionParamsType,
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  getActionType,
+  TorqActionType,
+} from './torq';
+
+import * as utils from './lib/axios_utils';
+
+jest.mock('axios');
+jest.mock('./lib/axios_utils', () => {
+  const originalUtils = jest.requireActual('./lib/axios_utils');
+  return {
+    ...originalUtils,
+    request: jest.fn(),
+    patch: jest.fn(),
+  };
+});
+
+axios.create = jest.fn(() => axios);
+const requestMock = utils.request as jest.Mock;
+
+axios.create = jest.fn(() => axios);
+
+const ACTION_TYPE_ID = '.torq';
+
+const services: Services = actionsMock.createServices();
+
+let actionType: TorqActionType;
+let mockedLogger: jest.Mocked<Logger>;
+
+beforeAll(() => {
+  const { logger, actionTypeRegistry } = createActionTypeRegistry();
+  actionType = actionTypeRegistry.get<
+    ActionTypeConfigType,
+    ActionTypeSecretsType,
+    ActionParamsType
+  >(ACTION_TYPE_ID);
+  mockedLogger = logger;
+});
+
+describe('actionType', () => {
+  test('exposes the action as `torq` on its Id and Name', () => {
+    expect(actionType.id).toEqual('.torq');
+    expect(actionType.name).toEqual('Torq');
+  });
+});
+
+describe('secrets validation', () => {
+  test('succeeds when secrets is valid', () => {
+    const secrets: Record<string, string> = {
+      token: 'jfi2fji3ofeaiw34if',
+    };
+    expect(validateSecrets(actionType, secrets)).toEqual(secrets);
+  });
+
+  test('fails when secret token is not provided', () => {
+    expect(() => {
+      validateSecrets(actionType, {});
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"error validating action type secrets: token is required"`
+    );
+  });
+});
+
+describe('config validation', () => {
+  const defaultValues: Record<string, string | null> = {};
+
+  test('config validation passes with an appropriate endpoint', () => {
+    const config: Record<string, string | boolean> = {
+      webhookIntegrationUrl: 'https://hooks.torq.io/v1/test',
+    };
+    expect(validateConfig(actionType, config)).toEqual({
+      ...defaultValues,
+      ...config,
+    });
+  });
+
+  ['iamnotavalidurl', 'example.com/do-something'].forEach((url) => {
```

Comment:
i would create an object looks like {url:string,error:string} and iterate it to reduce the number of tests like table tests in golang